tests/std/tests/VSO_0157762_feature_test_macros/test.cpp

```diff
@@ -472,1396 +438,1366 @@ STATIC_ASSERT(__cpp_structured_bindings == 201606L);
+#ifndef __cpp_hex_float
+#error Expected __cpp_hex_float to be defined.
+#elif __cpp_hex_float != 201603L
+#error Expected cpp_hex_float to equal 201603L.
 #else
-STATIC_ASSERT(__cpp_conditional_explicit == 201806L);
+STATIC_ASSERT(__cpp_hex_float == 201603L);
 #endif
 #else
-#ifdef __cpp_conditional_explicit
-#error BOOM
+#ifdef __cpp_hex_float
+#error Expected __cpp_hex_float to NOT be defined.
 #endif
 #endif
 
-#if CXX20_MODE && !defined(__clang__)
+#if _HAS_CXX20 && !defined(__clang__)
 // Clang only has partial support for <=> but that does not include the feature test macro.
 #ifndef __cpp_impl_three_way_comparison
-#error BOOM
+#error Expected __cpp_impl_three_way_comparison to be defined.
 #elif defined(__EDG__) // EDG does not yet implement P1630R1 or P1186R3 so they still report the old value.
 #if __cpp_impl_three_way_comparison != 201711L
-#error BOOM
+#error Expected cpp_impl_three_way_comparison to equal 201711L.
 #else
 STATIC_ASSERT(__cpp_impl_three_way_comparison == 201711L);
 #endif
 #else
 #if __cpp_impl_three_way_comparison != 201907L
-#error BOOM
+#error Expected cpp_impl_three_way_comparison to equal 201907L.
 #else
 STATIC_ASSERT(__cpp_impl_three_way_comparison == 201907L);
 #endif
 #endif
 #else
 #ifdef __cpp_impl_three_way_comparison
-#error BOOM
+#error Expected __cpp_impl_three_way_comparison to NOT be defined.
 #endif
 #endif
 
-// Conditionally defined (various compiler options) to specific values.
-
-// C++17, /Zc:alignedNew[-]
-#if !CXX17_MODE || defined(TEST_DISABLED_ALIGNED_NEW)
-#ifdef __cpp_aligned_new
-#error BOOM
+#if _HAS_CXX20 && !defined(__clang__) && !defined(__EDG__)
+#ifndef __cpp_nontype_template_args
+#error Expected __cpp_nontype_template_args to be defined.
+#elif __cpp_nontype_template_args != 201911L
+#error Expected cpp_nontype_template_args to equal 201911L.
+#else
+STATIC_ASSERT(__cpp_nontype_template_args == 201911L);
 #endif
+#elif _HAS_CXX17
+#ifndef __cpp_nontype_template_args
+#error Expected __cpp_nontype_template_args to be defined.
+#elif __cpp_nontype_template_args != 201411L
+#error Expected cpp_nontype_template_args to equal 201411L.
 #else
-#ifndef __cpp_aligned_new
-#error BOOM
-#elif __cpp_aligned_new != 201606L
-#error BOOM
+STATIC_ASSERT(__cpp_nontype_template_args == 201411L);
+#endif
 #else
-STATIC_ASSERT(__cpp_aligned_new == 201606L);
+#ifdef __cpp_nontype_template_args
+#error Expected __cpp_nontype_template_args to NOT be defined.
+#endif
+#endif
+
+#if _HAS_CXX20 || defined(__EDG__)
+// EDG unconditionally reports "explicit(bool)" as being available.
+#ifndef __cpp_conditional_explicit
+#error Expected __cpp_conditional_explicit to be defined.
+#elif __cpp_conditional_explicit != 201806L
+#error Expected cpp_conditional_explicit to equal 201806L.
+#else
+STATIC_ASSERT(__cpp_conditional_explicit == 201806L);
+#endif
+#else
+#ifdef __cpp_conditional_explicit
+#error Expected __cpp_conditional_explicit to NOT be defined.
 #endif
 #endif
 
 // C++98, /EHs[-]

```

Comment:
I sorted the sections using msort so these got sorted by their comments (except `//C++98...` which I manually adjusted). Should they be kept that way or sorted using the condition?

Reply:
IMO, they should be sorted lexicographically by the name under test.